### PR TITLE
Add diff chunking for large PR reviews

### PR DESCRIPTION
**Pull request by haacked**
## Summary

- Adds `chunk-diff.sh` that splits oversized diffs into logical chunks using directory proximity, test file pairing, and size balancing heuristics
- Integrates chunking into the review orchestrator and handler so agents process chunks independently
- Activates only when diff exceeds configurable thresholds (default 200KB or 50 files); small PRs are completely unaffected
- Synthesis step deduplicates findings across both agents and chunks

## Test plan

- [ ] 26 new unit tests for `chunk-diff.sh` cover: threshold behavior, test file pairing, directory grouping, size balancing, edge cases
- [ ] 4 new orchestrator tests verify chunking integration
- [ ] All 1032 existing tests still pass
- [ ] Manual test with a large PR (50+ files) to verify chunking activates and produces coherent reviews

### CODE REVIEW

File: skills/review-code/scripts/review-orchestrator.sh [hunk 1]
```diff
@@ -289,6 +289,50 @@ build_review_data() {
...
+    debug_time "05-chunking" "start"
+    local chunk_result=""
+    local chunk_max_size_kb="${REVIEW_CODE_CHUNK_MAX_SIZE_KB:-200}"
+    [[ "${chunk_max_size_kb}" =~ ^[0-9]+$ ]] || chunk_max_size_kb=200
+    local chunk_max_files="${REVIEW_CODE_CHUNK_MAX_FILES:-30}"
+    [[ "${chunk_max_files}" =~ ^[0-9]+$ ]] || chunk_max_files=30
+    local chunk_threshold_kb="${REVIEW_CODE_CHUNK_THRESHOLD_KB:-400}"
+    [[ "${chunk_threshold_kb}" =~ ^[0-9]+$ ]] || chunk_threshold_kb=400
+    local chunk_threshold_files="${REVIEW_CODE_CHUNK_THRESHOLD_FILES:-50}"
+    [[ "${chunk_threshold_files}" =~ ^[0-9]+$ ]] || chunk_threshold_files=50
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/review-orchestrator.sh:302]: Good catch. The 200KB in the PR description is the per-chunk size limit (`max_chunk_size_kb`), not the activation threshold. The activation threshold is 400KB (or 50 files). I'll fix the PR description.

File: skills/review-code/scripts/chunk-diff.sh [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    paired_groups=$(echo "${segments_json}" | jq -r --argjson meta "${file_metadata_json}" '
+        # Derive the implementation file path from a test file name.
+        # Returns null if the file does not match any known test pattern.
+        def impl_path_for_test($basename; $dirpart):
+            if ($basename | test("^test_")) then
+                ($dirpart + "/" + ($basename | sub("^test_"; "")))
+            elif ($basename | test("_test\\.go$")) then
+                ($dirpart + "/" + ($basename | sub("_test\\.go$"; ".go")))
+            elif ($basename | test("_test\\.")) then
+                ($dirpart + "/" + ($basename | sub("_test\\."; ".")))
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/chunk-diff.sh:168]: True, but test files at the repo root are vanishingly rare in practice. The fallback behavior is just "no pairing" (files stay unpaired and get chunked individually), not a crash. Not worth adding complexity for this edge case.

File: skills/review-code/scripts/chunk-diff.sh [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+                    (reduce $chunk.files[] as $f ("";
+                        . + ($seg_lookup[$f].diff // "")
+                    )) as $chunk_diff |
+
+                    # Calculate chunk size
+                    ($chunk_diff | length / 1024 | floor) as $size_kb |
+
+                    # Generate label from most common top-level directory
+                    ([$chunk.files[] | split("/")[0]] | group_by(.) | sort_by(-length) | .[0][0] // "root") as $top_dir |
+
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/chunk-diff.sh:326]: This is incorrect. jq's `group_by` sorts internally before grouping. From the jq docs: "group_by(.foo) takes as input an array, groups the elements having the same .foo field into separate arrays, and produces all of these arrays as elements of a larger array, sorted by the value of the .foo field."

Quick verification: `echo '["src","lib","src"]' | jq 'group_by(.) | sort_by(-length) | .[0][0]'` correctly returns `"src"`.

File: skills/review-code/scripts/chunk-diff.sh [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+    file_count="${file_count:-0}"
+
+    # Check if below threshold
+    local threshold_kb_bytes=$((min_threshold_kb * 1024))
+    if [[ "${diff_size_bytes}" -lt "${threshold_kb_bytes}" ]] && [[ "${file_count}" -lt "${min_threshold_files}" ]]; then
+        jq -n \
+            --arg reason "below threshold (${diff_size_kb}KB, ${file_count} files)" \
+            '{"chunked": false, "reason": $reason}'
+        return
+    fi
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/chunk-diff.sh:144]: Valid observation, but the cost is negligible in practice. The awk parsing on a sub-threshold diff (under 400KB/50 files) takes a few milliseconds. A `grep -c` pre-check would save that at the cost of an extra pass over the file. Since this code only runs once per review session, the optimization is not worth the added complexity.
